File: CalWatchAndroid/app/src/main/java/com/calwatch/android/api/models/FilterParams.java
package com.calwatch.android.api.models;

/**
 * Created by John R. Kosinski on 23/1/2559.
 */
public class FilterParams {
    private String dateFrom;
    private String dateTo;
    private String timeFrom;
    private String timeTo;

    public String getDateFrom() { return dateFrom;}
    public void setDateFrom(String dateFrom) {
        if (dateFrom == null || dateFrom.length() == 0)
            dateFrom = null;
        this.dateFrom = dateFrom;
    }

    public String getDateTo() {
        return dateTo;
    }
    public void setDateTo(String dateTo) {
        if (dateTo == null || dateTo.length() == 0)
            dateTo = null;
        this.dateTo = dateTo;
    }

    public String getTimeFrom() {
        return timeFrom;
    }
    public void setTimeFrom(String timeFrom) {

        if (timeFrom == null || timeFrom.length() == 0)
            timeFrom = null;
        this.timeFrom = timeFrom;
    }

    public String getTimeTo() {
        return timeTo;
    }
    public void setTimeTo(String timeTo) {

        if (timeTo == null || timeTo.length() == 0)
            timeTo = null;
        this.timeTo = timeTo;
    }

    public boolean isEmpty()
    {
        return ((dateFrom == null || dateFrom.length() == 0) &&
                (dateTo == null || dateTo.length() == 0) &&
                (timeFrom == null || timeFrom.length() == 0) &&
                (timeTo == null || timeTo.length() == 0)
        );
    }


    public FilterParams(){}

    public FilterParams(String dateFrom, String dateTo, String timeFrom, String timeTo)
    {
        this.setDateFrom(dateFrom);
        this.setDateTo(dateTo);
        this.setTimeFrom(timeFrom);
        this.setTimeTo(timeTo);
    }
}

File: CalWatchApi/CalWatch.Api/Models/FilterParams.cs
using System;

namespace CalWatch.Api.Models
{
    public class FilterParams
    {
        public DateTime DateFrom { get; private set; }
        public DateTime DateTo { get; private set; }
        public TimeSpan TimeFrom { get; private set; }
        public TimeSpan TimeTo { get; private set; }

        public FilterParams(
            DateTime? dateFrom = null,
            DateTime? dateTo = null,
            TimeSpan? timeFrom = null,
            TimeSpan? timeTo = null
        )
        {
            this.SetDateFrom(dateFrom);
            this.SetDateTo(dateTo);
            this.SetTimeFrom(timeFrom);
            this.SetTimeTo(timeTo);
        }

        public void SetDateFrom(DateTime? value)
        {
            if (value == null)
                value = DateTime.MinValue;

            this.DateFrom = value.Value;
        }

        public void SetDateTo(DateTime? value)
        {
            if (value == null)
                value = DateTime.MaxValue;

            this.DateTo = value.Value;
        }

        public void SetTimeFrom(TimeSpan? value)
        {
            if (value == null)
                value = TimeSpan.Parse("00:00:00");

            this.TimeFrom = value.Value;
        }

        public void SetTimeTo(TimeSpan? value)
        {
            if (value == null)
                value = TimeSpan.Parse("23:59:59");

            this.TimeTo = value.Value;
        }
    }
}
